srtcore/core.cpp

```diff
@@ -2688,24 +2764,28 @@ bool CUDT::interpretSrtHandshake(const CHandShake& hs,
     }
 
     if (hs.m_iVersion < HS_VERSION_SRT1)
-        return true; // do nothing
+        return false; // do nothing
 
-    // Anyway, check if the handshake contains any extra data.
-    if (hspkt.getLength() <= CHandShake::m_iContentSize)
+    // It's not necessary to check size to fit in the basic handshake,
+    // it's already done during handshake serialization.
+
+    // Now check the obligatory HS flags. HSX is required, KMX and CONFIG optional.
+    int ext_flags = SrtHSRequest::SRT_HSTYPE_HSFLAGS::unwrap(hs.m_iType);
```

Comment:
This is just inverted order, while the size check is for not just the plain handshake, but + the HSREQ extension. Basic size doesn't have to be checked - it was done already when deserializing the handshake. But then the flag should be checked first because with the flag it should grant that it also contains appropriate extensions.